nulib/meadow
# Make actions dropdown closeable.

# Summary 
First, a note. This ended up being a larger issue as the way we handled it required an integration of a new component from our design system. So apologies for the partial overhaul here.

This pull request make actions dropdown closeable by replacing it's existing functionality with a Popover component from the design-system repository. As a side effect of bringing in popover, we also needed to make some adjustments to the Technical Metadata and Delete (Work/Fileset) modals to address z-index layering issues. To do this, I have brought in the Dialog component from Radix UI. Long term a modal component should be created for the design system (issue created as [#2489](https://github.com/nulib/repodev_planning_and_docs/issues/2489)), but for now this has been inserted directly from the Radix UI package -- making the length of `Preservation.jsx` a bit longer than it could be.

![image](https://user-images.githubusercontent.com/7376450/144442014-ae09815b-f984-4c52-b190-2edf291eb468.png)

# Specific Changes in this PR
- Swap out actions menu for design-system popover.
- Convert technical metadata modal to Radix UI alert dialog.
# Version bump required by the PR

See [Semantic Versioning 2.0.0](https://semver.org/) for help discerning which is required.


# Steps to Test

1.  Go to a work with multiple filesets associated.
2. Click Preservation tab
3. Click actions, close the panel by clicking the `X` close button, hitting `esc`, or by clicking outside of the popover pane.
4. Click View technical metadata, a modal should appear, which is also closeable. The modal should disable interaction with outside elements and only popover content will be visible to screen readers until closed.
5. Can delete work and filesets still

# :rocket: Deployment Notes

**Note** - __if you check any of these boxes go to the [(always open) `main` <- `staging` PR](https://github.com/nulib/meadow/pulls) and add detailed notes and instructions to help out others who may end up deploying your changes to production__

- Backward compatible API changes
- Backwards-incompatible API changes


# Tested/Verified
